modules/renter/contractor/contractor.go

```diff
@@ -84,6 +84,31 @@ func (c *Contractor) Contract(hostAddr modules.NetAddress) (modules.RenterContra
 	return modules.RenterContract{}, false
 }
 
+// PeriodSpending returns the amount spent on contracts during the current
+// billing period.
+func (c *Contractor) PeriodSpending() modules.ContractorSpending {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	spending := modules.ContractorSpending{}
+	allSpending := types.ZeroCurrency
+	for _, contract := range c.contracts {
+		spending.ContractSpending = spending.ContractSpending.Add(contract.TotalCost)
+		spending.DownloadSpending = spending.DownloadSpending.Add(contract.DownloadSpending)
+		spending.UploadSpending = spending.UploadSpending.Add(contract.UploadSpending)
+		spending.StorageSpending = spending.StorageSpending.Add(contract.StorageSpending)
+		for _, pre := range contract.PreviousContracts {
+			spending.ContractSpending = spending.ContractSpending.Add(pre.TotalCost)
+			spending.DownloadSpending = spending.DownloadSpending.Add(pre.DownloadSpending)
+			spending.UploadSpending = spending.UploadSpending.Add(pre.UploadSpending)
+			spending.StorageSpending = spending.StorageSpending.Add(pre.StorageSpending)
+		}
+		allSpending = allSpending.Add(spending.ContractSpending).Add(spending.DownloadSpending).Add(spending.UploadSpending).Add(spending.StorageSpending)
```

Comment:
Does this end up with a heavy over-estimate for `allSpending`? I think you want to do this final addition outside of the contract loop, otherwise you are double-adding all previous spending every contract.